### PR TITLE
ktg5's Pull Request.

### PR DESCRIPTION
**Pull request by ktg5**
I've added some new commands, ping command actually get's it's ping, edited the `user` command a bit, and added random colors to most embed messages.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,6 +42,16 @@ async function uuid(username){
     return response;
 }
 
+var special_ids = {
```
Comment (james090500) [inline index.js:45]: There ID's should be for Mccapes staff roles

478670065483513860 //support staff
479048180202340362 //contributor
628211166321311744 //cape creative

File: index.js [hunk 2]
```diff
@@ -44,52 +78,119 @@ async function checkUrl(url){
...
+    try {
+        if (command == `help`) {
+            var randomColor = Math.floor(Math.random()*16777215).toString(16);
+            var sender = message.member.user;
+
+            var DMreply = await embed(`List of commands`, `\`${prefix}help\` - Helps you\n\`${prefix}ping\` - Pings the bot\n\`${prefix}user (mc-username)\` - Gets info about the user`, randomColor)
+            sender.createDM().then (dmChannel => {
+                dmChannel.send(DMreply)
+            }).catch (err => {
+                message.channel.send(`I was unable to do finish this command, make sure your DMs are open to all server members. \`ERR: 0xCR0003\``)
```
Comment (james090500) [inline index.js]: Typo here

File: index.js [hunk 3]
```diff
@@ -44,52 +78,119 @@ async function checkUrl(url){
...
+            message.channel.send(reply);
         }
-
-        var description = '**[NameMC](https://namemc.com/profile/' + response.long_id + ')**';
-        var thumbnail = 'https://crafatar.com/avatars/' + response.id + '?overlay=true';
-        var color = '#21ff33'
-        var reply = await embed(args[0], description, color, fields, thumbnail);
-        message.channel.send(reply);
+    } catch (err) {
+        var reply = embed(`Oops! I just got a error.`, `I guess report it to staff, here's the error I got: \n` + "```" + err + "```", `0xFF9900`)
```
Comment (james090500) [inline index.js]: Typo here

"I just got an error"

File: index.js [hunk 4]
```diff
@@ -44,52 +78,119 @@ async function checkUrl(url){
...
+            var cape_urls = config.cape_urls;
+    
+            if (!args[0]) {
+                var reply = await embed(`User error!`, `You didn't enter a username after the command!`, `0xFF0000`)
+                message.channel.send(reply)
+                return;
             }
+    
+            if(response === null) {
+                var reply = await embed("Invalid Username!", "The username is invalid, please make sure you typed it in correctly.", `0xFF0000`);
```
Comment (james090500) [inline index.js:157]: Typo here

"please make sure you typed it in correctly."

File: index.js [hunk 5]
```diff
@@ -44,52 +78,119 @@ async function checkUrl(url){
...
+            var DMreply = await embed(`List of commands`, `\`${prefix}help\` - Helps you\n\`${prefix}ping\` - Pings the bot\n\`${prefix}user (mc-username)\` - Gets info about the user`, randomColor)
+            sender.createDM().then (dmChannel => {
+                dmChannel.send(DMreply)
+            }).catch (err => {
+                message.channel.send(`I was unable to finish this command, make sure your DMs are open to all server members. \`ERR: 0xCR0003\``)
+                return
+            });
+            var reply = await embed(`:mailbox_with_mail:\`You got mail!\``, `${new Date().getTime() - message.createdTimestamp}ms`, randomColor)
+            message.channel.send(reply);
+        }
```
Comment (Siriuo) [inline index.js]: This will need to be removed or replaced with a different command as it conflicts with other bots.

**Review by Siriuo — CHANGES_REQUESTED** [review]

File: index.js (hunk 5 above)
Comment (ktg5) [reply index.js]: How so?

Comment (Siriuo) [reply index.js]: It conflicts with the MEE6 bot and its !help command.

Comment (ktg5) [reply index.js]: Could just change the prefix to something else.

Comment (ktg5) [reply index.js]: Or I got a better idea.

**Pull request merged**